File: android/android-snippets/app/src/main/java/com/google/example/mlkit/LanguageIdentificationActivity.java
package com.google.example.mlkit;

import android.os.Bundle;
import android.util.Log;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.google.android.gms.tasks.OnFailureListener;
import com.google.android.gms.tasks.OnSuccessListener;
import com.google.mlkit.nl.languageid.IdentifiedLanguage;
import com.google.mlkit.nl.languageid.LanguageIdentification;
import com.google.mlkit.nl.languageid.LanguageIdentificationOptions;
import com.google.mlkit.nl.languageid.LanguageIdentifier;

import java.util.List;


public class LanguageIdentificationActivity extends AppCompatActivity {

    private static final String TAG = "LangID";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    private void identifyLanguageWithStringInput(String text) {
        // [START identify_languages]
        LanguageIdentifier languageIdentifier =
                LanguageIdentification.getClient();
        languageIdentifier.identifyLanguage(text)
                .addOnSuccessListener(
                        new OnSuccessListener<String>() {
                            @Override
                            public void onSuccess(@Nullable String languageCode) {
                                if (languageCode.equals("und")) {
                                    Log.i(TAG, "Can't identify language.");
                                } else {
                                    Log.i(TAG, "Language: " + languageCode);
                                }
                            }
                        })
                .addOnFailureListener(
                        new OnFailureListener() {
                            @Override
                            public void onFailure(@NonNull Exception e) {
                                // Model couldn’t be loaded or other internal error.
                                // ...
                            }
                        });
        // [END identify_languages]
    }

    private void setConfidence() {
        // [START set_confidence]
        LanguageIdentifier languageIdentifier = LanguageIdentification.getClient(
                new LanguageIdentificationOptions.Builder()
                        .setConfidenceThreshold(0.34f)
                        .build());
        // [END set_confidence]
    }

    private void getPossibleLanguuages(String text) {
        // [START get_possible_languages]
        LanguageIdentifier languageIdentifier =
                LanguageIdentification.getClient();
        languageIdentifier.identifyPossibleLanguages(text)
                .addOnSuccessListener(new OnSuccessListener<List<IdentifiedLanguage>>() {
                    @Override
                    public void onSuccess(List<IdentifiedLanguage> identifiedLanguages) {
                        for (IdentifiedLanguage identifiedLanguage : identifiedLanguages) {
                            String language = identifiedLanguage.getLanguageTag();
                            float confidence = identifiedLanguage.getConfidence();
                            Log.i(TAG, language + " (" + confidence + ")");
                        }
                    }
                })
                .addOnFailureListener(
                        new OnFailureListener() {
                            @Override
                            public void onFailure(@NonNull Exception e) {
                                // Model couldn’t be loaded or other internal error.
                                // ...
                            }
                        });
        // [END get_possible_languages]
    }

    private void setConfidenceThreshold() {
        // [START set_confidence_threshold]
        LanguageIdentificationOptions identifierOptions =
                new LanguageIdentificationOptions.Builder()
                        .setConfidenceThreshold(0.5f)
                        .build();
        LanguageIdentifier languageIdentifier = LanguageIdentification
                .getClient(identifierOptions);
        // [END set_confidence_threshold]
    }
}

File: android/android-snippets/app/src/main/java/com/google/example/mlkit/kotlin/LanguageIdentificationActivity.kt
package com.google.example.mlkit.kotlin

import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import com.google.mlkit.nl.languageid.LanguageIdentification
import com.google.mlkit.nl.languageid.LanguageIdentificationOptions

private const val TAG = "LangIDActivity"

class LanguageIdentificationActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
    }

    fun identifyLanguageWithStringInput(text: String) {
        // [START identify_languages]
        val languageIdentifier = LanguageIdentification.getClient()
        languageIdentifier.identifyLanguage(text)
                .addOnSuccessListener { languageCode ->
                    if (languageCode == "und") {
                        Log.i(TAG, "Can't identify language.")
                    } else {
                        Log.i(TAG, "Language: $languageCode")
                    }
                }
                .addOnFailureListener {
                    // Model couldn’t be loaded or other internal error.
                    // ...
                }
        // [END identify_languages]
    }

    fun setConfidence() {
        // [START set_confidence]
        val languageIdentifier = LanguageIdentification
                .getClient(LanguageIdentificationOptions.Builder()
                        .setConfidenceThreshold(0.34f)
                        .build())
        // [END set_confidence]
    }

    fun getPossibleLanguuages(text: String) {
        // [START get_possible_languages]
        val languageIdentifier = LanguageIdentification.getClient()
        languageIdentifier.identifyPossibleLanguages(text)
                .addOnSuccessListener { identifiedLanguages ->
                    for (identifiedLanguage in identifiedLanguages) {
                        val language = identifiedLanguage.languageTag
                        val confidence = identifiedLanguage.confidence
                        Log.i(TAG, "$language $confidence")
                    }
                }
                .addOnFailureListener {
                    // Model couldn’t be loaded or other internal error.
                    // ...
                }
        // [END get_possible_languages]
    }

    private fun setConfidenceThreshold() {
        // [START set_confidence_threshold]
        val identifierOptions = LanguageIdentificationOptions.Builder()
                .setConfidenceThreshold(0.5f)
                .build()
        val languageIdentifier = LanguageIdentification
                .getClient(identifierOptions)
        // [END set_confidence_threshold]
    }


}
